docs/design/README.md

```diff
@@ -0,0 +1,595 @@
+It is also possible to generate these spec files as they would be simple aggregations of all the packages in the kit.
+
+### Kit Structure
+
+A kit container will be built using `FROM scratch` because it does not need a runtime environment.
+It's directory structure will look like this:
+
+```sh
+# tree /local --dirsfirst
+/local
+├── etc
+│   └── yum.repos.d
+│       └── bottlerocket-core-kit.repo
+└── kits
+    └── bottlerocket-core-kit
+        ├── archives
+        │   └── bottlerocket-x86_64-kernel-5.15-archive-5.15.102-1.x86_64.rpm
+        ├── repodata
+        │   ├── 30e098715e13cbe4733c6b4518dcc2cdc68d1ac3277eefad2ee5ac171c4b9023-primary.sqlite.bz2
+        │   ├── 5a22c22832fa81e55f84de715cf910cebc64d472be65a7124fd8b02ff7c777fb-filelists.xml.gz
+        │   ├── 719d1eef5ce4d194d4c5c51171dce8987cb21e88338f1d5dff7554dacabf1004-other.sqlite.bz2
+        │   ├── 7cbab1a2f05c1eda86d3d125497af680d9100665e3eb30c7560926eabbc2c655-primary.xml.gz
+        │   ├── d61fe62ca166eab1cbd502cf21cd977798ef348de49fb35e38a93ebf696eb0a5-filelists.sqlite.bz2
+        │   ├── fcd4dadc59c6d6512410b103a38762bf266389e01a60add12319191c4cc16ec4-other.xml.gz
+        │   └── repomd.xml
+        ├── bottlerocket-x86_64-acpid-2.0.34-1.x86_64.rpm
+        ├── bottlerocket-x86_64-apiclient-0.0-0.x86_64.rpm
+        ├── ...
+        └── bottlerocket-x86_64-wicked-0.6.68-1.x86_64.rpm
+```
+
+The `/etc/yum.repos.d` file can be used later to configure `dnf`.
+This will allow `dnf` to treat each kit as a yum repository.
+When aggregating kits, the yum repos will be prioritized according to the maintainer's requirements.
+
+### Kit Dependencies
+
+One kit may depend on a package from another kit.
+All kits used together, need to have been built by the same SDK.
+Another way to say this is that all kits must depend on the same SDK.
+These dependencies need to be specified at the kit level so that tooling understands which kits are required for a build.
+
+We will create a metadata container image where these dependencies are enumerated in a JSON file.
+Consider a kit named `my-awesome-kit`.
+This may be published to a container registry and tagged with versions by the publisher:
+
+```
+registry.com/my-awesome-kit-x86_64:v0.1.0
+registry.com/my-awesome-kit-x86_64:v0.2.0
+etc.
+```
+
+Consider that a package within this kit depends on Bottlerocket's `glibc`.
+Our JSON metadata for `my-awesome-kit` will let tell us about the kit and SDK dependencies:
+
+```json
+{
+  "kit": {
+    "name": "my-awesome-kit",
+    "version": "v0.1.0",
+    "arch": "x86_64",
+    "sdk": "public.ecr.aws/bottlerocket/bottlerocket-sdk-x86_64:v0.50.0",
+    "dependencies": [
+        "public.ecr.aws/bottlerocket/bottlerocket-core-kit-x86_64:v1.15.1"
+    ]
+  }
+}
+```
+
+In this example, it would be an error if `bottlerocket-core-kit-x86_64:v1.15.1` declared that it had been built with anything other `public.ecr.aws/bottlerocket/bottlerocket-core-kit-x86_64:v1.15.1`.
```

Comment:
```suggestion
In this example, it would be an error if `bottlerocket-core-kit-x86_64:v1.15.1` declared that it had been built with anything other `public.ecr.aws/bottlerocket/bottlerocket-sdk-x86_64:v0.50.0`.
```